go.mod

```diff
@@ -2,17 +2,30 @@ module github.com/absmach/coap-cli
 
 go 1.22
 
-require github.com/plgd-dev/go-coap/v2 v2.6.0
+require (
+	github.com/plgd-dev/go-coap/v3 v3.3.3
+	github.com/spf13/cobra v1.8.0
+)
+
```

Comment:
Rearrange (combine the indirect imports) then run `go mod tidy` also updated `github.com/plgd-dev/go-coap/v3`